src/app/components/weather-alert/weather-alert.component.ts

```diff
@@ -1,22 +1,35 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
 
 import {  WeatherService } from 'src/app/services/weather.service';
 import { WeatherAlertResponse } from 'src/app/models/weather-alert.model';
+import { interval, Subscription, take } from 'rxjs';
 
 @Component({
   selector: 'app-weather-alert',
   templateUrl: './weather-alert.component.html',
   styleUrls: ['./weather-alert.component.scss']
 })
-export class WeatherAlertComponent implements OnInit {
+export class WeatherAlertComponent implements OnInit, OnDestroy {
   @Input() weatherAlert: WeatherAlertResponse;
   public showToast: boolean = true; // By default we want to show the toast, but will be changed if the user closes the toast or if new weather alert data is called from the API
+  public toastTimeoutBarWidth: number = 100; // Default the bar to 100% width
+  private toastTimeoutTime: number = 10000; // Adjustable to make the toast display a shorter or longer amount of time
+  private toastTimeoutUpdateInterval: number = 10; // Adjustable to make the shrinking of the timer bar smoother in exchange for more processing power. The smaller the number, the more frequently the bar width will be calculated.
+  private toastLifespanEndTime: number;
+  private timeoutBarUpdateIntervalSubscription: Subscription;
 
   constructor(private weatherService: WeatherService) { }
 
   ngOnInit(): void {
     this.subscribeToWeatherService(); // subscribe to updates in the weatherService so we can show the toast in case it was closed
     this.timerToRemoveAlert();
+    this.toastLifespanEndTime = new Date().getTime() + this.toastTimeoutTime;
```

Comment:
can this be moved into `setupIntervalTimer()` so its not created in `ngOnInit()`

Reply:
Yes, good catch